internal/utils/templates/templates.go

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright 2023 Nikolai Rodionov (allanger)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package templates
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"html/template"
+	"strings"
+
+	"github.com/db-operator/db-operator/api/v1beta1"
+	"github.com/db-operator/db-operator/pkg/consts"
+	"github.com/db-operator/db-operator/pkg/utils/database"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/utils/strings/slices"
+)
+
+const (
+	DEFAULT_TEMPLATE        = "{{ .Protocol }}://{{ .Username }}:{{ .Password }}@{{ .Hostname }}:{{ .Port }}/{{ .Database }}"
+	DEFAULT_NAME            = "CONNECTION_STRING"
+	TEMPLATE_ANNOTATION_KEY = "kinda.rocks/db-operator-templated-keys"
+)
+
+func (tds *TemplateDataSources) BuildVars(templates v1beta1.Templates) error {
```

Comment:
would mostly split this function. 80 lines is hard to read/understand.

Reply:
Ok, I'll try splitting, and also I wanted to rename it to "Render" or something

Reply:
I've tried refactoring it, I guess it looks better now, I still would like to avoid that duplicating logic for Secrets and ConfigMaps, but currently I don't know how